packages/projects-docs/pages/learn/integrations/github-app.md

```diff
@@ -33,6 +33,58 @@ After the installation request is made, the owner of the repository or an admin
 - Installation request that appears when importing a repo for the first time.
 - GitHub alls can be installed on an organization level or for individual repositories
 
+## Setting up preview links
+The GitHub App reads `tasks.json` for preview links. You can add links to the PR description by adding the following fields to the tasks that you want run as a preview:
+
+```
+"preview": {
+  "port": 3000,
+  "pr-link": "direct" | "redirect" | "devtool"
+  }
+```
+
+In the list of tasks it would look something like this
+```
+"dev": {
+  "name": "dev",
+  "command": "yarn dev",
+  "runAtStart": true,
+  "preview": {
+    "port": 4000,
+    "pr-link": "direct"
+  }
+},
+"build": {
+  ...
+}
+  ```
+  
+Note: If you do not have a `tasks.json` file, you can see instructions [here](../setting-up/tasks#opening-configuration) on how to open the tasks configuration.
+
+### port: Optional
+This field allows you to determine which port will be used for the preview. This is great for ensuring that the preview url is always the same. If this is not listed as a field, the preview will open on any available port which my be different each time.
+
+### pr-link: Required
```

Comment:
At the moment, the backend requires `port` and makes `pr-link` optional. We should discuss if that's the desired behaviour.

Reply:
Another note: do we want to use `prLink`? It's currently `pr-link` because that was in a preliminary document, but I don't think we ever formally decided that.

Reply:
prLink is preferable because we use camel case for `runAtStart`. 
I'd like to make sure the docs are accurate though so if that is something we can change soon, then we can hold off on publishing the docs till that is updated in the code.

Reply:
> At the moment, the backend requires port and makes pr-link optional. We should discuss if that's the desired behaviour.
Thanks for clarifying. I was pulling from notes in [this ticket](https://linear.app/codesandbox/issue/CSB-4714/add-standalone-preview-link-to-pr-descriptions)

> Q: How do users turn off the preview links altogether?
> A: By not having the "pr-link" key in the "preview" configuration

My understanding from the loom in the ticket is that if there is no port listed, it is possible for us to choose an available one, is that right? If so, it makes sense to me that the port is optional and the pr-link (or prLink) is required, unless we have a default for that, in which case we could look at make that optional. 

If they are both potentially optional, then a follow-up question might be: could we make all the fields optional and then have the `preview` field accept an object or a boolean and make the check for a preview be: if `preview = true` create preview with default ports and links, if `preview is object`, use requirements listed in the obj?

Reply:
> My understanding from the loom in the ticket is that if there is no port listed, it is possible for us to choose an available one, is that right?

I'm not sure on this. The server wouldn't know what port to choose, but is it okay if the preview link doesn't include a port at all? If the client knows what to do, we could leave off the `?port=...` parameter for the default and redirect links. Not sure what the direct link would look like though.

Reply:
Thumbs up on `prLink`, sorry for even suggesting `pr-link`, weird 👍 

I think we should make `port` required. The reason is that in principle we can not guarantee what port to open on a process as the process can open multiple ports. So making this a requirement does not only fix the links, but also when we automatically open the preview in the Project Editor.

We should bring @alexnm into this conversation as he worked on the tasks and automatic preview opening.

It makes sense in the context of the PR LINK to make port optional, which makes "direct" link invalid... but seems we end up with a more consistent and more accessible feature by just making it a requirement? 🤔